Source/DafnyCore/Resolver/PreType/PreTypeResolve.Statements.cs

```diff
@@ -929,6 +941,17 @@ public partial class PreTypeResolver : INewOrOldResolver {
         return;
       }
 
+      if (s.Proof != null) {
+        // clear the labels for the duration of checking the proof body, because break statements are not allowed to leave the proof body
```

Comment:
What about revealing labelled assertions inside the proof? Would be good to add a test for those